### PR TITLE
add Python 3.13 and Django 5.1 support + remove deprecated versions

### PR DESCRIPTION
**Pull request by rupertbaker**

### CODE REVIEW

**Review by carltongibson — COMMENTED** [review]
Looks sensible

**Comment by carltongibson** [comment]
Well, if you're actual do due diligence... 😅

Not sure how to smooth that. Maybe a warnings test run with `-Werror`, but beyond that, yes you'd need to read. 

I guess that gives you a nice touch point though, so I won't feel too bad. 

🎁

**Comment by aaugustin** [comment]
I'm quite thorough with these things because I don't like dead code :-)

File: tox.ini [hunk 1]
```diff
@@ -1,21 +1,17 @@
 [tox]
 isolated_build = true
 envlist =
-    py38-django32
-    py310-django40
-    py311-django41
+    py39-django42
     py312-django42
```
Comment (aaugustin) [inline tox.ini:5]: Should have been changed to django50. I'm fixing it in the next commit.

**Pull request merged**